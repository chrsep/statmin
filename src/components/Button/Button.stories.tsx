import { action } from "@storybook/addon-actions"
import { storiesOf } from "@storybook/react"
import * as React from "react"
import { darkTheme } from "../../theme"
import Button from "./Button"

storiesOf("Button", module)
  .add("with text", () => (
    <Button onClick={action("clicked")} theme={darkTheme}>
      Grant Access
    </Button>
  ))
  .add("with some emoji", () => (
    <Button onClick={action("clicked")} theme={darkTheme}>
      <span role="img" aria-label="so cool">
        😀 😎 👍 💯
      </span>
    </Button>
  ))
