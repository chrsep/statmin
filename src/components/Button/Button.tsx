import styled from "../../styled-components"

export default styled.button`
  color: ${({ theme }) => theme.white};
  border-color: ${({ theme }) => theme.accentColor};
  font-weight: bold;
  background: transparent;
  border-style: solid;
  border-width: 3px;
  border-radius: 10rem;
  padding: 0.8rem;
  transition: all 0.2s;
  :hover {
    background-color: ${({ theme }) => theme.accentColor};
    color: ${({ theme }) => theme.black};
  }
`
