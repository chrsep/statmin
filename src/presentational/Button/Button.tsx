import styled from "../../styled-components"

export default styled.button`
  color: ${({ theme }) => theme.white};
  border-color: ${({ theme }) => theme.accentColor};
  font-weight: bold;
  font-size: 1.6rem;
  background: transparent;
  border-style: solid;
  border-width: 2px;
  border-radius: 10rem;
  padding: 0.8rem;
  transition: all 0.1s;
  outline: none;
  user-select: none;
  :active {
    transform: scale(1.1);
  }
`
