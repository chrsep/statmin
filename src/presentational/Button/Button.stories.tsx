import * as React from "react"
import { action } from "storybook__addon-actions"
import { storiesOf } from "storybook__react"
import { darkTheme } from "../../theme"
import Button from "./Button"

storiesOf("Button", module)
  .add("with text", () => (
    <Button onClick={action("clicked")} theme={darkTheme}>
      Grant Access
    </Button>
  ))
  .add("with some emoji", () => (
    <Button onClick={action("clicked")} theme={darkTheme}>
      <span role="img" aria-label="so cool">
        😀 😎 👍 💯
      </span>
    </Button>
  ))
