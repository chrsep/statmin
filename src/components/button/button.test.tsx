import * as React from "react"
import { mount } from "enzyme"
import Button from "./button"

describe("Button Component", () => {
  it("renders correctly", () => {
    mount(<Button />)
  })
})
