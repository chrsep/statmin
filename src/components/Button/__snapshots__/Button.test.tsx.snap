// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders correctly 1`] = `
.c0 {
  color: white;
  border-color: #4fe6c4;
  font-weight: bold;
  font-size: 1.6rem;
  background: transparent;
  border-style: solid;
  border-width: 2px;
  border-radius: 10rem;
  padding: 0.8rem;
  -webkit-transition: all 0.1s;
  transition: all 0.1s;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0:active {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

<button
  className="c0"
/>
`;
